### DISCUSSION TITLE
P/S wave marker problem

### DISCUSSION THREAD

**Discussion by woxin5295** [SeismicSource/sourcespec#34] [Q&A] [answered]
Hi，everybody ，
I enter the P arrival and S arrival times into the SAC header, using the symbols A and T0, respectively.
The SAC default marker T0 to P wave arrival time, and it automatically adds T0 symbols, like this:

      OMARKER = -1.5259e-06
      AMARKER = 115.27           (T0)
     T0MARKER = 209.01

One consequence of this situation is that the SourceSpec code cannot find the P wave arrival time. How can this problem be fixed?

Junqing

**Comment by claudiodsf** [comment, marked as answer]
Hi Junqing,

Your `A` field has for some reason the label `"T0"` (the label for `A` is stored in the SAC header field `KA`). This is non-standard, since the label `"T0"` should go with the header field `T0`.

SourceSpec identifies the picks using:
- the field labels, if they are defined
- or assuming that `A` is the P pick and `T0` is the S pick, if no label is defined.

So, in this case, SourceSpec reads `"T0"` as the `A` field label, and it doesn't know how to identify this pick.

There are two possible solutions to this:

1. you remove the labels and let SourceSpec assume that `A` is the P pick and `T0` is the S pick, using, e.g., the following SAC command:

       SAC> ch ka undef kt0 undef

2. you explicitly mark your picks as `"P"` and `"S"`, using, e.g., the following SAC command:

       SAC> ch ka P kt0 S

**Reply by claudiodsf** [reply]
I improved the README and the [online documentation](https://sourcespec.readthedocs.io/en/latest/file_formats.html#phase-pick-formats) to better explain how to specify phase picks in the SAC header.

**Reply by woxin5295** [reply]
Many thanks, Prof. claudiodsf. This problem was solved.
